core/primitives/src/beacon.rs

```diff
@@ -26,13 +28,33 @@ pub struct BeaconBlock {
     pub header: BeaconBlockHeader,
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct SignedBeaconBlock {
     pub body: BeaconBlock,
     pub hash: CryptoHash,
     pub signature: GroupSignature,
 }
 
+impl Borrow<CryptoHash> for SignedBeaconBlock {
+    fn borrow(&self) -> &CryptoHash {
+        &self.hash
+    }
+}
+
+impl Hash for SignedBeaconBlock {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(self.hash.as_ref());
```

Comment:
`CryptoHash` implements `hash` therefore here we can use `state.hash(self.hash)` which is more consistent, because it does not re-hash the array of bytes. Same for other places.

Reply:
Fixed